libraries/cyclestreets-fragments/src/main/java/net/cyclestreets/CycleMapFragment.kt

```diff
@@ -0,0 +1,150 @@
+package net.cyclestreets
+
+import android.Manifest
+import net.cyclestreets.fragments.R
+
+import net.cyclestreets.views.CycleMapView
+
+import org.osmdroid.views.overlay.Overlay
+
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import net.cyclestreets.util.Logging
+
+import net.cyclestreets.util.MenuHelper.createMenuItem
+import net.cyclestreets.util.MenuHelper.enableMenuItem
+import net.cyclestreets.util.doOrRequestPermission
+import android.content.pm.PackageManager
+import android.preference.PreferenceManager
+import org.osmdroid.config.Configuration
+import org.osmdroid.config.DefaultConfigurationProvider
+import java.io.File
+import java.util.Date
+
+private val TAG = Logging.getTag(CycleMapFragment::class.java)
+
+open class CycleMapFragment : Fragment(), Undoable {
+    private var map: CycleMapView? = null
+    private var forceMenuRebuild: Boolean = false
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, saved: Bundle?): View? {
+        super.onCreate(saved)
+
+        forceMenuRebuild = true
+
+        checkPermissionNoMoreThanOnceEveryFiveMinutes()
```

Comment:
Because of the fragment lifecycle, if we don't do this then at the time of starting up the fragment we enter an infinite-request loop.  I think we should still ask for the permission each *meaningful* time we navigate to a map fragment (of whatever type) - hence this mechanism, rather than e.g. only asking at app startup.